source/DSCResources/DSC_CMDistributionPoint/DSC_CMDistributionPoint.psm1

```diff
@@ -0,0 +1,615 @@
+
+        [Parameter()]
+        [ValidateSet('Http','Https')]
+        [String]
+        $ClientCommunicationType = 'Http',
+
+        [Parameter()]
+        [String[]]
+        $BoundaryGroups,
+
+        [Parameter()]
+        [ValidateSet('Add','Remove','Match')]
+        [String]
+        $BoundaryGroupStatus = 'Add',
+
+        [Parameter()]
+        [Boolean]
+        $AllowPreStaging,
+
+        [Parameter()]
+        [Boolean]
+        $EnableAnonymous,
+
+        [Parameter()]
+        [Boolean]
+        $EnableBranchCache,
+
+        [Parameter()]
+        [Boolean]
+        $EnableLedbat,
+
+        [Parameter()]
+        [ValidateSet('Present','Absent')]
+        [String]
+        $Ensure = 'Present'
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+    $state = Get-TargetResource -SiteCode $SiteCode -SiteServerName $SiteServerName
+
+    try
+    {
+        if ($Ensure -eq 'Present')
+        {
+            if ($state.Ensure -eq 'Absent')
+            {
+                if (($PrimaryContentLibraryLocation.Length -gt 1 -or $PrimaryContentLibraryLocation -match '[0-9]') -or
+                   ($SecondaryContentLibraryLocation.Length -gt 1 -or $SecondaryContentLibraryLocation -match '[0-9]') -or
+                   ($PrimaryPackageShareLocation.Length -gt 1 -or $PrimaryPackageShareLocation -match '[0-9]') -or
+                   ($SecondaryPackageShareLocation.Lenth -gt 1 -or $SecondaryPackageShareLocation -match '[0-9]'))
+                {
+                    throw $script:localizedData.InvalidPriOrSecLetter
+                }
+
+                if (($SecondaryContentLibraryLocation -and [string]::IsNullOrEmpty($PrimaryContentLibraryLocation)) -or
+                   ($SecondaryPackageShareLocation -and [string]::IsNullOrEmpty($PrimaryPackageShareLocation)))
+                {
+                    throw $script:localizedData.SecAndNoPrimary
+                }
+
+                if ($null -eq (Get-CMSiteSystemServer -SiteCode $SiteCode -SiteSystemServerName $SiteServerName))
+                {
+                    Write-Verbose -Message ($script:localizedData.SiteServerRole -f $SiteServerName)
+                    New-CMSiteSystemServer -SiteCode $SiteCode -SiteSystemServerName $SiteServerName
+                }
+
+                $initialValues = @('MinimumFreeSpaceMB','PrimaryContentLibraryLocation','SecondaryContentLibraryLocation',
+                                    'PrimaryPackageShareLocation','SecondaryPackageShareLocation')
+
+                foreach ($item in $initialValues)
+                {
+                    if ($PSBoundParameters.ContainsKey($item))
+                    {
+                        $dpSetupParams += @{
+                            $item = $PSBoundParameters.$item
+                        }
+                    }
+                }
+
+                if ($PSBoundParameters.ContainsKey('CertificateExpirationTimeUtc'))
```

Comment:
It could be if it was specified, if it wasn't specified it would create Cert expiration time for you. i could add the param piece up top and if (-not $PSBoundParameters.ContainsKey('CertificateExpirationTimeUtc')  then create the param with 2 years.